src/components/RightSidebar/SearchMessages/SearchMessagesTab.vue

```diff
@@ -0,0 +1,442 @@
+	resourceUrl: string
+	icon: string
+	rounded: boolean
+	attributes: MessageSearchResultAttributes
+	to: Route
+}
+
+const emit = defineEmits<{
+	(event: 'close'): void
+}>()
+
+const searchBox = ref(null)
+const isFocused = ref(false)
+const searchResults = ref<MessageSearchResultEntry[]>([])
+const searchText = ref('')
+const fromUser = ref<UserFilterObject | null>(null)
+const sinceDate = ref<Date | null>(null)
+const untilDate = ref<Date | null>(null)
+const searchLimit = ref(10)
+const searchCursor = ref(0)
+const searchDetailsOpened = ref(false)
+const isFetchingResults = ref(false)
+const isSearchExhausted = ref(false)
+
+const store = useStore()
+const isInCall = useIsInCall()
+
+const token = computed(() => store.getters.getToken())
+const participantsInitialised = computed(() => store.getters.participantsInitialised(token.value))
+const participants = computed<UserFilterObject>(() => {
+	return store.getters.participantsList(token.value)
+		.map(({ actorId, displayName, actorType }: { actorId: string; displayName: string; actorType: string}) => ({
+			id: actorId,
+			displayName,
+			isNoUser: actorType !== 'users',
+			user: actorId,
+			disableMenu: true,
+			showUserStatus: false,
+		}))
+})
+const canLoadMore = computed(() => !isSearchExhausted.value && !isFetchingResults.value && searchCursor.value !== 0)
+const hasFilter = computed(() => fromUser.value || sinceDate.value || untilDate.value)
+
+onMounted(() => {
+	EventBus.on('route-change', onRouteChange)
+})
+
+onBeforeUnmount(() => {
+	EventBus.off('route-change', onRouteChange)
+	abortSearch()
+})
+
+const onRouteChange = ({ from, to }: { from: Route, to: Route }): void => {
+	if (to.name !== 'conversation' || from.params.token !== to.params.token || (to.hash && isInCall.value)) {
+		abortSearch()
+		emit('close')
+	}
+}
+
+watch(searchText, (value) => {
+	if (value.trim().length === 0) {
+		searchResults.value = []
+		searchCursor.value = 0
+		isSearchExhausted.value = false
+	}
+})
+
+/**
+ * Cancel search and cleanup the search fields and results.
+ */
+function abortSearch() {
+	cancelSearch.value('canceled')
+	searchText.value = ''
+	fromUser.value = null
+	sinceDate.value = null
+	untilDate.value = null
+	searchDetailsOpened.value = false
+	searchResults.value = []
+	searchCursor.value = 0
+}
+
+/**
+ * Continue fetching more search results
+ */
+function loadMore() {
+	return fetchSearchResults(false)
+}
+
+/**
+ *
+ */
+function fetchNewSearchResult() {
+	return fetchSearchResults(true)
+}
+
+const cancelSearch = ref((cancel: string) => {})
+
+/**
+ * @param [isNew=true] Is it a new search (search parameters changed)?
+ * Fetch the search results from the server
+ */
+async function fetchSearchResults(isNew = true) {
+	// Don't search if the search text is empty
+	if (searchText.value.trim().length === 0) {
+		return
+	}
+
+	isFetchingResults.value = true
+
+	try {
+		cancelSearch.value('canceled')
+		const { request, cancel } = CancelableRequest(searchMessages)
```

Comment:
Atm it's not made correctly, but we could use 'as' for requests. Check NewMessageChatSummary.vue for reference

```suggestion
type SearchMessageCancelableRequest = {
	request: (payload: SearchMessagePayload) => UnifiedSearchResponse,
	cancel: () => void,
}

		const { request, cancel } = CancelableRequest(searchMessages) as SearchMessageCancelableRequest
```